ksqldb-execution/src/main/java/io/confluent/ksql/execution/plan/KTableHolder.java

```diff
@@ -88,6 +88,13 @@ public LogicalSchema getSchema() {
     return new KTableHolder<>(table, schema, executionKeyFactory, materializationBuilder);
   }
 
+  public KTableHolder<K> withTable(
+          final KTable<K, GenericRow> table,
+          final LogicalSchema schema,
+          final Optional<MaterializationInfo.Builder> builder) {
+    return new KTableHolder<>(table, schema, executionKeyFactory, builder);
```

Comment:
I think the reason is to inherit the `executionKeyFactory` from the parent while all others are overridden.

I'm thinking that since we are replacing all others except the `executionKeyFactory`, maybe we should just remove this function and on the caller just call

```
materialized(table, schema, parent.getExecutionKeyFactory(), materializedBuilder)
```

instead?